### PR TITLE
fix: badge should accept negative number

### LINKED ISSUE #15799: Badge Bug
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/embed/5kpp8jzkyp)

### Steps to reproduce
the problem is when I pass a negative number into Badge Component that give me the same number but with positive value , i mean the component don't care about negative values 

### What is expected?
i expect the component give me negtive values in display 

### What is actually happening?
the component instead of add  ( - )  signal , it puts 0 

| Environment | Info |
|---|---|
| antd | 3.16.1 |
| React | react-dom (version 16.8.6) |
| System | linux ubuntu |
| Browser | yes |

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

fix #15799

`-10` will display as `010`:
屏幕快照 2019-04-02 下午3 03 36

### 💡 Solution

Check number. Not use scroll number if is negative.

### 📝 Changelog

Fix Badge display negative count incorrect.

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/badge/ScrollNumber.tsx [hunk 1]
```diff
@@ -132,7 +132,8 @@ class ScrollNumber extends Component<ScrollNumberProps, ScrollNumberState> {
 
   renderNumberElement(prefixCls: string) {
     const { count } = this.state;
-    if (count && Number(count) % 1 === 0) {
+    const numVal = Number(count);
+    if (count && numVal % 1 === 0 && numVal >= 0) {
```
Comment (ztplz) [inline components/badge/ScrollNumber.tsx]: 那不是把负数当字符处理了吗  滚动动画没了

Comment (zombieJ) [reply components/badge/ScrollNumber.tsx]: 现在的动画只支持从 0 到正数效果：https://ant.design/components/badge-cn/#components-badge-demo-change

 到 0 后就自动消失。

**Comment by ztplz** [comment]
https://github.com/ant-design/ant-design/blob/master/components/badge/ScrollNumber.tsx#L106
改这里怎么样  
``````
isNaN(position) ? "-" : i % 10
``````

**Comment by zombieJ** [comment]
@ztplz ，目前动画是到 0 为止的。小于 0 的数字，渲染 number list 意义不大。直接当 string 处理就够了。

**Comment by ztplz** [comment]
Untitled7

**Comment by zombieJ** [comment]
有道理，Why not？ 我改一下

**Comment by zombieJ** [comment]
本来想做成动画翻滚方向统一，发现实现太困难，就简单做了。
@afc163 @ztplz

**Review by afc163 — APPROVED** [review]

**Comment by knanahassouna1** [comment]
thanks

**Pull request merged**